web/src/admin/files/FileUploadForm.ts

```diff
@@ -0,0 +1,126 @@
+import "#elements/forms/HorizontalFormElement";
+
+import { DEFAULT_CONFIG } from "#common/api/config";
+import { MessageLevel } from "#common/messages";
+
+import { Form } from "#elements/forms/Form";
+import { PreventFormSubmit } from "#elements/forms/helpers";
+import { showMessage } from "#elements/messages/MessageContainer";
+
+import { AdminApi, AdminFileListUsageEnum } from "@goauthentik/api";
+
+import { msg } from "@lit/localize";
+import { html } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import { createRef, ref } from "lit/directives/ref.js";
+
+@customElement("ak-file-upload-form")
+export class FileUploadForm extends Form<Record<string, unknown>> {
+    @property({ type: String, useDefault: true })
+    public usage: AdminFileListUsageEnum = AdminFileListUsageEnum.Media;
+
+    @state()
+    protected selectedFile: File | null = null;
+
+    #formRef = createRef<HTMLFormElement>();
+
+    #fileChangeListener = (e: Event) => {
+        const input = e.target as HTMLInputElement;
+        if (input.files?.length) {
+            this.selectedFile = input.files[0];
+        } else {
+            this.selectedFile = null;
+        }
+    };
+
+    protected clearFileInput() {
+        this.selectedFile = null;
+        this.#formRef.value?.reset();
+    }
+
+    public override async send(data: Record<string, unknown>): Promise<void> {
+        if (!this.selectedFile) {
+            throw new PreventFormSubmit("Selected file not provided", this);
+        }
+
+        // Same regex is used in the backend as well
+        const ValidFilenameRegex = /^[a-zA-Z0-9._/-]+$/;
```

Comment:
Recommend either:

- Use something like `kebabCase` to convert the file to the preferred format
- Moving this into a const outside the class as `VALID_FILE_NAME_PATTERN`